### ISSUE TITLE
Question about YOLOv8 License

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [huggingface/candle#959]
Hi there,

thanks a lot for this amazing project! Since yolov8 here is reimplemented, we can use it commercially right? The original GPL license is for the code and the pretrained weights

Thanks a lot

**Comment by LaurentMazare** [comment]
Hello,
Not a lawyer and I don't have much of a clue of whether this could be used commercially or not. What I can say is that:
- This implementation is based on the [tinygrad version](https://github.com/tinygrad/tinygrad/blob/master/examples/yolov8.py) so subject probably to the same restriction that applies for this one.
- The model is kept intentionnaly in its own crate/example directory and not moved with the other shared models in `candle-transformers` so as to avoid tainting the whole crate if the GPL license applied here.

**Comment by FrancescoSaverioZuppichini** [comment]
> Hello,
> 
> Not a lawyer and I don't have much of a clue of whether this could be used commercially or not. What I can say is that:
> 
> - This implementation is based on the [tinygrad version](https://github.com/tinygrad/tinygrad/blob/master/examples/yolov8.py) so subject probably to the same restrictions that applies for this one.
> 
> - The model is kept intentionnaly in its own crate/example directory and not moved with the other shared models in `candle-transformers` so as to avoid tainting the whole crate if the GPL license applied here.

Thanks a lot for the quick reply 🤗 I'll dig more into it, I am 99% sure that if you reimplement it and retrain then GPL doesn't apply anymore since you are not using their previous code/weights

**Comment by LaurentMazare** [comment]
> Thanks a lot for the quick reply 🤗 I'll dig more into it, I am 99% sure that if you reimplement it and retrain then GPL doesn't apply anymore since you are not using their previous code/weights

Ah one good thing to point out then is that this model is not retrained and uses the weight files provided by the tinygrad version. My guess is that they just converted the initial checkpoints.

**Comment by FrancescoSaverioZuppichini** [comment]
> > Thanks a lot for the quick reply 🤗 I'll dig more into it, I am 99% sure that if you reimplement it and retrain then GPL doesn't apply anymore since you are not using their previous code/weights
> 
> Ah one good thing to point out then is that this model is not retrained and uses the weight files provided by the tinygrad version. My guess is that they just converted the initial checkpoints.

well then I guess we just need the loss function and some GPU hours do make a MIT YOLOv8 - which would be huge 

Closing now 

Thanks again :)

Fra
